api/envoy/config/route/v4alpha/route_components.proto

```diff
@@ -710,6 +736,8 @@ message RouteAction {
 
     // Determines if upgrades are available on this route. Defaults to true.
     google.protobuf.BoolValue enabled = 2;
+
+    ConnectConfig connect_config = 3;
```

Comment:
I think this came up in Slack, but would this make more sense as part of this oneof in Route? https://github.com/envoyproxy/envoy/blob/2413b833c796066cdc84318384d3626aa6002afe/api/envoy/config/route/v3/route_components.proto#L209-L224.

I can see the logical benefit of it being part of `RouteAction`, but many of the fields don't apply (some do like hash policy I guess). I think this is fine but I just want to make sure we are all on the same page. cc @htuch

Reply:
Also, if connect_config is not specified (wherever it ends up) the connect will be refused, right? I would clarify that.

Reply:
I'm fine with this being called `ConnectAction` and placed here. I thought that @alyssawilk mentioned that there is some overlapping state between this and `RouteAction`, it would be good to not duplicate too much of this is a significant overlap.

Reply:
So I had a note to talk to you about that then opted to take it to the PATH pr, since it's not relevant until we remove the synthetic path.  
HTTP/2 allows connect-with-path.  If there's connect with PATH should we allow the connect matchers to match it or still refuse?
It's the difference between "CONNECT requests will not be matched" on the other matchers and "requests without a PATH (such as HTTP/1 CONNECT requests) will not be matched.

Reply:
> CONNECT requests will not be matched" on the other matchers

This is how I assumed it would work. Meaning, if h2 supports connect with path we can still do a header match on path, but by having the connect matcher it is much more intentional what is being matched. I'm still willing to be swayed that this split does not make sense, but given how much of a special case CONNECT is and all of the inherent security implications IMO it's not a bad split to have. wdyt?

Reply:
Yeah, I was much of the same mind - CONNECT is so dangerous I'm super twitchy about enabling folks to screw themselves up.  That said, CONNECT doesn't work by default, and you have to configure it 2 places for HTTP/2 so I think if we comment it would be Ok to allow patch matches.  I'd prefer to only do it if folks need it, but given the 2 year deprecation dance I guess we have to guess.  Allowing it does mean we couldn't validate that we only allow ConnectConfig with connect matchers, since then any matcher could work as a connect matcher, but still the most dangerous form of connect (terminating and forwarding payload) would only happen on routes which explicitly have ConnectConfig, so there's enough checks I think it'd be hard to do it by accident.

Reply:
Basically I could go either way, but if there's a path and 3 guards I guess I'm inclined to let the patch matchers work.

Reply:
(though while folks are thinking about the pros and cons, it's a bit weird that a .* regex matcher wouldn't match if there's no path header.  I think as long as we comment explicitly that they only work with Path present it's OK.)